### ISSUE TITLE
关于博客文章的疑惑

### ISSUE THREAD

**Issue by raymond8825** [babyking/KarabinerConfiguration#4]
您之前的文章:https://babyking.github.io/wiki/%E5%8D%9A%E5%AE%A2%E5%A4%87%E4%BB%BD/2019-07-18-ru-he-li-jie-karabiner-de-zhe-ji-ge-shi-jian-/,

<img width="1389" alt="image" src="https://user-images.githubusercontent.com/41637672/216937699-cb04ab99-9663-41b5-8646-3245fbb579f3.png">

我有一些困惑,这里黄框是不是to和if_alone并用以及if_alone和if_held_down ?想问一下这里是不是笔误?

**Comment by babyking** [comment]
时间太久了,我有些记不清细节了,但这句话的场景还是比较清楚的.

键盘上的26个字母有很多我设置了长按键设置,比如"F"键,我把其长按设置成了command, 正常按还是"F",开始我是这样实现的
```json
{
    "from": {
        "key_code": "f",
        "modifiers": {
            "optional": [
                "any"
            ]
        }
    },
    "to": [
        {
            "key_code": "command_left"
        }
    ],
    "to_if_alone": [
        {
            "key_code": "f"
        }
    ],
    "type": "basic"
}
```


.但会有个问题, 我打字速度比较快,有时候经常前一个字母按下还没有充分抬起,第二个字母我已经按下了. 这样会导致字母f无法触发,还会小概率出现f直接触发了command.  后面我好像又使用held_down和alone配置使用,依然会有问题.直到最终是通过下面这样的配置来解决的.

```json
{
  "from": {
      "key_code": "f",
      "modifiers": {
          "mandatory": []
      }
  },
  "to_delayed_action": {
      "to_if_canceled": [
          {
              "key_code": "f"
          }
      ]
  },
  "to_if_alone": [
      {
          "halt": true,
          "key_code": "f"
      }
  ],
  "to_if_held_down": [
      {
          "key_code": "left_command"
      }
  ],
  "type": "basic"
}
```


to_delayed_action  就是用来解决上面说的前一个字母还没充分抬起,后一个字母(同时)已经按下的情况.

如果你没有我遇到的问题,就可以忽略.

**Comment by raymond8825** [comment]
是的,您这种情况我之前确实遇到了,我是用了上述的方法,但是发现有一个问题:就是必须得按键超过我们设定的阈值,而我们打字的速度较快,往往也会出现问题,比如我把f映射成commad键盘,打字快了,很难适应需要长按f才能触发comma的问题,后来我又用回了原来的配置,没有把26个字母映射成修饰键了,现在映射了"," 和"/"按键,用来配合yabai和hammerspoon,因为我的小拇指长按cpas和大拇指长按空格都已经有点不舒服了.无论如何,感谢您的文章.另外,单击shift切换输入法实在是很好用

**Comment by babyking** [comment]
每个人习惯不同,需要定制.   asdf四个键映射到修饰键主要也是为了解决在打字过程中偶尔需要使用快捷键而不离开打字键位,并不是用来替代原有快捷键. 在打字场景下一般也不需要很快的触发快捷键,所以这个长按阈值是可以接受的. 但第一需要保证的是单击打字的100%准确性.  谢谢反馈.
